src/config.rs

```diff
@@ -0,0 +1,191 @@
+//! Implementation of the configuration module.
+
+use anyhow::Context;
+use anyhow::Result;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::Mode;
+
+/// Represents the configuration for the Sprocket CLI tool.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct Config {
+    /// Configuration for the `format` command.
+    #[serde(rename = "format")]
+    pub format_config: FormatConfig,
+    /// Configuration for the `check` and `lint` commands.
+    #[serde(rename = "check")]
+    pub check_config: CheckConfig,
+    /// Common configuration options for all commands.
+    #[serde(rename = "common")]
+    pub common_config: CommonConfig,
+}
+
+/// Represents shared configuration options for Sprocket commands.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct CommonConfig {
+    /// Display color output.
+    pub color: bool,
+    /// The report mode.
+    pub report_mode: Option<Mode>,
+}
+
+/// Represents the configuration for the Sprocket `format` command.
+#[derive(Debug, Clone, Default, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case", deny_unknown_fields)]
+pub struct FormatConfig {
+    /// Disables color output.
+    pub no_color: bool,
+    /// Enable or disable color output.
+    pub color: bool,
+    /// The report mode.
+    pub report_mode: Option<Mode>,
```

Comment:
this should be removed, right? if they are in the `CommonConfig` they shouldn't be here as well IMO